src/components/Modal/ExtensionStatusModal.tsx

```diff
@@ -0,0 +1,226 @@
+            </div>
+        );
+    }
+
+    return (
+        <div
+            className={styles.extensionModalOverlay}
+            onClick={handleOutsideClick}
+            data-testid="extension-modal-overlay"
+        >
+            <div
+                className={styles.extensionModal}
+                ref={modalRef}
+                data-testid="extension-modal-content"
+            >
+                <h2 data-testid="modal-title">Extension Details</h2>
+
+                {extensionRequests.length > 0 ? (
+                    extensionRequests.map((request) => (
+                        <div
+                            key={request.id}
+                            className={styles.extensionExtensionRequest}
+                            data-testid={`extension-request-${request.id}`}
+                        >
+                            {getExtensionRequestDetails(request, styles).map(
+                                (item, idx) => (
+                                    <div
+                                        key={idx}
+                                        className={styles.extensionDetailRow}
+                                        data-testid={`detail-row-${idx}`}
+                                    >
+                                        <span
+                                            className={styles.extensionLabel}
+                                            data-testid={`label-${item.testId}`}
+                                        >
+                                            {item.label}
+                                        </span>
+                                        <span
+                                            className={`${
+                                                styles.extensionValue
+                                            } ${item.className || ''}`}
+                                            data-testid={`value-${item.testId}`}
+                                        >
+                                            {item.value}
+                                        </span>
+                                    </div>
+                                )
+                            )}
+
+                            {request.reviewedBy && (
+                                <div
+                                    className={styles.extensionApprovalInfo}
+                                    data-testid="approval-info"
+                                >
+                                    Your request was approved by{' '}
+                                    {request.reviewedBy}{' '}
+                                    {request.reviewedAt
+                                        ? formatToRelativeTime(
+                                              request.reviewedAt
+                                          )
+                                        : ''}
+                                </div>
+                            )}
+                        </div>
```

Comment:
- please create a component that takes in the extension requests and renders either the empty state "no extension requests found" or iterates on the extension requests and renders the details.